src/cargo/util/toml/mod.rs

```diff
@@ -2225,10 +2226,12 @@ fn to_dependency_source_id<P: ResolveToPath + Clone>(
 pub(crate) fn lookup_path_base<'a>(
     base: &PathBaseName,
     gctx: &GlobalContext,
-    workspace_root: &dyn Fn() -> CargoResult<&'a PathBuf>,
-    features: &Features,
+    workspace_root: &dyn Fn() -> CargoResult<&'a Path>,
+    features: Option<&Features>,
```

Comment:
This feels sketchy to make optional.  We still need people to do the check somewhere

Reply:
Is there a good way to get `Features` from within `cargo add`? I didn't see any other use of it, and trying to construct one manually turned into a bit of a mess.